mlir/lib/Conversion/GPUToGEN/GPUToGEN.cpp

```diff
@@ -0,0 +1,139 @@
+//===- GPUToGEN.cpp - GPU to GEN Patterns ----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements patterns to convert GPU dialect to GEN dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/GPUToGEN/GPUToGEN.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/GEN/IR/GENDialect.h"
+#include "mlir/Dialect/GEN/IR/GENOps.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "llvm/Support/Debug.h"
+
+namespace mlir {
+#define GEN_PASS_DEF_CONVERTGPUOPSTOGENOPS
+#include "mlir/Conversion/Passes.h.inc"
+} // namespace mlir
+
+using namespace mlir;
+
+template <typename GPUOp, typename GENOp>
+class GPUIndexOpToGENLowering : public OpConversionPattern<GPUOp> {
+public:
+  using OpConversionPattern<GPUOp>::OpConversionPattern;
+  using OpAdaptor = typename GPUOp::Adaptor;
+
+  LogicalResult
+  matchAndRewrite(GPUOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto dim = static_cast<std::uint32_t>(adaptor.getDimension());
+    Value idxDim = rewriter.create<arith::ConstantOp>(
+        op->getLoc(), rewriter.getI32IntegerAttr(dim));
```

Comment:
```suggestion
    Value idxDim = rewriter.create<arith::ConstantIntOp>(op->getLoc(), dim, 32);
```

NIT: I think this should work and is a shortcut for the above. If it works, apply in other places.

Reply:
That works but there was nowhere else to apply it. There is no specialization for a bool that I could find.

Reply:
(bool is just `i1`, but LGTM as is)